pallets/aleph/src/lib.rs

```diff
@@ -0,0 +1,103 @@
+//! Aleph session manager.
+//!
+//! This pallet manages the changes in the committee responsible for establishing consensus.
+//! Currently, it's PoA where the validators are set by the root account. In the future, a new
+//! pallet for PoS elections will replace this one.
+//!
+//! For full integration with Aleph finality gadget, the `primitives::AlephSessionApi` should be implemented.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+mod migrations;
+
+use sp_std::prelude::*;
+
+use frame_support::{
+    sp_runtime::BoundToRuntimeAppPublic,
+    traits::{OneSessionHandler, StorageVersion},
+    Parameter,
+};
+pub use pallet::*;
+
+/// The current storage version.
+const STORAGE_VERSION: StorageVersion = StorageVersion::new(2);
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use frame_support::{pallet_prelude::*, sp_runtime::RuntimeAppPublic};
+    use frame_system::pallet_prelude::BlockNumberFor;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        type AuthorityId: Member
+            + Parameter
+            + RuntimeAppPublic
+            + Default
+            + MaybeSerializeDeserialize;
+    }
+
+    #[pallet::pallet]
+    #[pallet::storage_version(STORAGE_VERSION)]
+    pub struct Pallet<T>(_);
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        fn on_runtime_upgrade() -> frame_support::weights::Weight {
+            migrations::v1_to_v2::migrate::<T, Self>()
```

Comment:
I'm pretty certain this is wrong. The `v0_to_v1` migration is not on chain yet. Why do we even consider removing migrations?

Reply:
Since it doesn't compile and I don't know how to force it. There is this tricky macro `generate_storage_alias!` that I used in `v1_to_v2` to remove fields of pallet aleph that doesn't exist in the new version, but here it's different since field `Validators` is of type `Vec<T::AccountId>` and I cannot use `T` in this macro.

We should then apply updates one by one.

Reply:
Applying updates one by one is of course an option, but we should have a discussion about it, as that would me a major change of our update strategy and essentially everything that is being done now considering devnet, featurnets, etc. 

I don't have experience in migrations, but you might ask Filip if he knows how to do that.

Reply:
This is fixed now.